bitwarden_license/src/Sso/Controllers/AccountController.cs

```diff
@@ -163,21 +173,39 @@ IActionResult invalidJson(string errorMessageKey, Exception ex = null)
         [HttpGet]
         public async Task<IActionResult> Login(string returnUrl)
         {
-            var context = await _interaction.GetAuthorizationContextAsync(returnUrl);
-            if (context.Parameters.AllKeys.Contains("domain_hint") &&
-                !string.IsNullOrWhiteSpace(context.Parameters["domain_hint"]))
+            try
             {
-                return RedirectToAction(nameof(ExternalChallenge), new
+                var redirectToken = Request.Cookies[SsoRedirectToken.SSO_REDIRECT_COOKIE_NAME];
+                if (string.IsNullOrEmpty(redirectToken))
```

Comment:
we're going to need to think about reverse compatibility with older clients that may not be able to handle the token... for those scenarios where it's not present, I can understand where the cookie approach was attractive, however as mentioned it won't work for Desktop, CLI or Mobile and may present issues with the Browser Extension as well, so it would only reliably work for Web.

This may require more careful design/consideration to work around or to ditch this particular approach altogether 🙁  or come up with a "phased" implementation that accounts for some level of reverse compatibility in validation.